driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java

```diff
@@ -102,69 +72,88 @@ private static boolean nameMatches(final String name, final String... prefixes)
         return false;
     }
 
-    @Nullable
-    static BsonDocument createClientMetadataDocument(@Nullable final String applicationName) {
-        return createClientMetadataDocument(applicationName, null);
-    }
-
-    @Nullable
-    public static BsonDocument createClientMetadataDocument(@Nullable final String applicationName,
+    public static BsonDocument getClientMetadataDocument(@Nullable final String applicationName,
```

Comment:
This was returning a cached document, so it was no longer being created each time.

This caching seems to have broken tests, so I removed it, for now. I suspect the caching is needed, because now with more client info, we need to repeatedly check the client info size, which seems more expensive than things were previously.

Reply:
Caching was used to minimize the cost of creating the metadata each time.